plugins/glustershd/rest.go

```diff
@@ -1,197 +1,129 @@
-	tmp := *v
-	oldvolinfo := &tmp
+	defer unlock(ctx)
 
-	// validate volume type
-	if !isVolReplicate(v.Type) {
-		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Volume Type not supported")
-		return
-	}
-
-	// Transaction which checks if all replicate volumes are stopped before
-	// stopping the self-heal daemon.
-	txn := transaction.NewTxn(ctx)
-	defer txn.Cleanup()
-
-	// Lock on volume name.
-	lock, unlock, err := transaction.CreateLockSteps(volname)
+	// Validate volume existence
+	volinfo, err := volume.GetVolume(volname)
 	if err != nil {
-		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		if err == gderrors.ErrVolNotFound {
+			logger.WithError(err).WithField(
+				"volname", volname).Debug("volume not found")
+			restutils.SendHTTPError(ctx, w, http.StatusNotFound, err)
+		} else {
+			logger.WithError(err).WithField(
+				"volname", volname).Debug("error occured while looking for volume")
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		}
 		return
 	}
 
-	v.HealEnabled = false
-
-	txn.Nodes = v.Nodes()
-	txn.Steps = []*transaction.Step{
-		lock,
-		{
-			DoFunc:   "vol-option.UpdateVolinfo",
-			Nodes:    []uuid.UUID{gdctx.MyUUID},
-			UndoFunc: "selfheald-undo",
-		},
-
-		{
-			DoFunc: "selfheal-stop",
-			Nodes:  txn.Nodes,
-		},
-		{
-			DoFunc: "vol-option.NotifyVolfileChange",
-			Nodes:  txn.Nodes,
-		},
-		unlock,
+	// Validate volume type
+	if !isVolReplicate(volinfo.Type) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "volume type is not supported")
+		return
 	}
 
-	if err := txn.Ctx.Set("volinfo", v); err != nil {
-		logger.WithError(err).Error("failed to set volinfo in transaction context")
-		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+	// Validate volume state
+	if volinfo.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrVolNotStarted)
 		return
 	}
 
-	if err := txn.Ctx.Set("oldvolinfo", oldvolinfo); err != nil {
-		logger.WithError(err).Error("failed to set volinfo in transaction context")
-		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+	if getHealInfo(volname, option) != nil {
+		logger.WithField("volname", volname).Debug("heal info operation failed")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "heal info operation failed")
 		return
 	}
 
-	err = txn.Do()
+	output := []byte(healInfoOutput)
+
+	var info glustershdapi.HealInfo
+	err = xml.Unmarshal(output, &info)
 	if err != nil {
-		logger.WithFields(log.Fields{
-			"error":   err.Error(),
-			"volname": volname,
-		}).Error("failed to stop self heal daemon")
+		logger.WithError(err).Error("Error unmarshalling from XML")
 		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
 		return
 	}
 
-	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, &info.Bricks)
```

Comment:
While defining rest routes, you have defined `glustershdapi.HealInfo` as the response type. That's the one that should be returned to the client.

Reply:
This hasn't been addressed.